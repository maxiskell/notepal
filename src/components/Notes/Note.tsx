import React from "react";

import { INote } from "../../redux/types";

const Note: React.FC<INote> = (props) => (
  <div className="card w-100 mt-4">
    <div className="card-body">
      <h5 className="card-title">{props.title}</h5>
      <p className="card-text" style={{ whiteSpace: "pre-wrap" }}>
        {props.content}
      </p>
    </div>
  </div>
);

export default Note;
