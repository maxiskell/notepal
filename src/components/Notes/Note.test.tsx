import React from "react";
import { render } from "@testing-library/react";
import Note from "./Note";

const mockNote = {
  id: 1,
  title: "Test Note",
  content: "Test Content",
};

test("render properly", () => {
  const { container } = render(<Note {...mockNote} />);

  expect(container).toMatchSnapshot();
});
