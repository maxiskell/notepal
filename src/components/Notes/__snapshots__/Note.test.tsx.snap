// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render properly 1`] = `
<div>
  <div
    class="card w-100 mt-4"
    data-testid="note-1"
    style="cursor: pointer;"
    title="Click to edit"
  >
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Test Note
      </h5>
      <p
        class="card-text"
        style="white-space: pre-wrap;"
      >
        Test Content
      </p>
    </div>
  </div>
</div>
`;
